docs/library/machine.CAN.rst

```diff
@@ -0,0 +1,364 @@
+    - ``f_clock`` is the system clock frequency for the CAN peripheral. This
+      can be omitted on many ports where the value is known to the hardware.
+
+    In this case, the CAN controller is configured correctly for the desired
+    ``bitrate`` and the specified total number of time quanta per bit.
+
+.. method:: CAN.init_fd(bitrate, sample_point=None, sjw=None, tseg1=None, tseg2=None, f_clock=None)
+
+   Initialise CAN FD for controllers which support it.
+
+   .. note:: This method is not present on the class for controllers that only
+             support CAN Classic.
+
+   This must be called after :func:`CAN.init()`. The parameters all have the
+   same meanings and behaviour as that function, however they configure the
+   Bit Rate Switch (BRS) feature of CAN FD.
+
+   When specifying exact timing parameters, the accepted ranges are:
+
+   - ``sjw`` can be a value between 1 and 16 inclusive.
+   - ``tseg1`` can be a value between 1 and 32 inclusive
+   - ``tseg2`` can be a value between 1 and 32 inclusive
+
+.. method:: CAN.detect_bitrate([timeout_ms])
+
+   Detect the CAN bit rate by monitoring the CAN bus.
+
+   ..note:: This method is only present on the class for controllers that
+     support this feature.
+
+   If successful, returns the detected bit rate and also configures the hardware
+   CAN controller for this rate.
+
+   If the optional ``timeout`` parameter is specified then detection only runs
+   for this many milliseconds before timing out and returning ``None``. In this
+   case the CAN controller configuration will remain unchanged after returning.
+
+.. method:: CAN.set_filters(filters)
+
+   Set receive filters in the CAN controller. ``filters`` should be an iterable,
+   where each item is a tuple or list with three elements:
+
+   - ``identifier`` is a CAN identifier (int).
+   - ``bit_mask`` is a bit mask for bits in the CAN identifier field (int).
+   - ``flags`` is one or more bitwise OR-ed values from
+     :class:`CAN.MessageFlags` that the incoming message needs to match. Not
+     all controllers support filtering on all flags, a ``ValueError`` is raised
+     if an unsupported flag is requested.
+
+  Incoming messages are accepted if the bits masked in ``bit_mask`` match between
+  the message identifier and the filter ``identifier`` value, and flags set in the
+  filter match the incoming message.
+
+  All filters are ORed together in the controller. Passing an empty list (``[]``)
+  for the filters argument disables the receive filter (all messages received.)
+
+   .. note:: If the caller passes a list with more entries than :data:`CAN.filters_max`,
+      ``ValueError`` will be raised.
+
+.. data:: CAN.filters_max
+
+   Constant value that reads the maximum number of supported receive filters
+   for this hardware controller.
+
+.. method:: CAN.send(identifier, data, flags=0, fifo_equal=True)
+
+   - ``identifier`` is an integer CAN identifier value.
```

Comment:
arg order is reversed from pyb can.

Reply:
Yeah... I find the pyb argument order here very unsettling for some reason, but I guess it's worth keeping it consistent. :)

Reply:
Went back and forth on this but decided to keep it. The remaining arguments are complete incompatible with `pyb.CAN`, so if someone copy-pastes code from there then they're going to have a bad time anyway. At least this way it's more likely to  produce an obvious error quicker, rather than more subtle incorrect behaviour.